modules/images/webp-uploads/helper.php

```diff
@@ -84,19 +83,23 @@ function webp_uploads_generate_additional_image_source( $attachment_id, $image_s
 	 * @return array|null|WP_Error An array with the file and filesize if the image was created correctly otherwise a WP_Error
 	 */
 	$image = apply_filters( 'webp_uploads_pre_generate_additional_image_source', null, $attachment_id, $image_size, $size_data, $mime );
-
 	if ( is_wp_error( $image ) ) {
 		return $image;
 	}
 
 	if (
-		is_array( $image )
-		&& ! empty( $image['file'] )
-		&& ! empty( $image['path'] )
+		is_array( $image ) &&
+		! empty( $image['file'] ) &&
+		(
+			! empty( $image['path'] ) ||
+			! empty( $image['filesize'] )
+		)
 	) {
 		return array(
 			'file'     => $image['file'],
-			'filesize' => filesize( $image['path'] ),
+			'filesize' => ! empty( $image['filesize'] )
+				? $image['filesize']
+				: filesize( $image['path'] ),
```

Comment:
Probably personal preference but I rather swap the condition here to remove the `!` operator.

```suggestion
			'filesize' => empty( $image['filesize'] )
				? filesize( $image['path'] )
				: $image['filesize'],
```

Another thing (unlikely) is the fact that `empty` would return `false` even when the key is present for `filesize` but the value is `0` or not truthy. I think would be safer to use either `isset` or `array_key_exists` instead.

```suggestion
			'filesize' => isset( $image['filesize'] )
				? $image['filesize']
				: filesize( $image['path'] ),
```

Or

```suggestion
			'filesize' => array_key_exists( $image,'filesize' )
				? $image['filesize']
				: filesize( $image['path'] ),
```

Reply:
Yep. Makes sense. Updated.